### ISSUE TITLE
Multi-agent control scenarios

### ISSUE THREAD

**Issue by sunningshore** [eleurent/highway-env#261]
Hi, thanks for the work. 

Just one small question regarding multi-agent control scenarios. When controlling a group of vehicles rather than one, it is usually desired that the initial positions of controled vehicles are close to each other, like a swarm of fleet. 

So, just wondering if it is possible to config initial positions of controlled vehicles? 

And also, _crash = True_ when controlled vehicles collide with each other. 

Thanks very much!

**Comment by eleurent** [comment]
Hi @sunningshore 
You may have a look at https://github.com/eleurent/highway-env/issues/260 which was a related question about initial speed.
TLDR: there is unfortunately no easy way to do that for now, as the initial scene is environment dependent and I find it a bit difficult to make it fully configurable. Possible solutions are:
1. you clone the project and edit the _create_vehicles() method of your favorite environment to place the vehicles wherever you like.
2. you manually set the position of each vehicle right after calling `env.reset()`, something like:
```
env.reset()
for vehicle in env.controlled_vehicles:
    vehicle.position = np.array([<desired x, desired y>])
```
But you need to to it everytime you reset the environment, and make sure you don't place a the vehicles near an obstacle, so I'd recommend solution 1.

> And also, crash = True when controlled vehicles collide with each other.

Ooops, is this not the case already? I'll look into that.

**Comment by sunningshore** [comment]
Hi Edouard, 

 

Thanks for your kind reply. This can certainly solve the problem. 

 “And also, crash = True when controlled vehicles collide with each other.

Ooops, is this not the case already? I'll look into that.”

Sorry, I have no idea. Because the 2 vehicles I created are always far from each other. But it’s good if it is already the case. 

Best!

**Comment by sunningshore** [comment]
Hi Edouard, 

 

It worked, thanks. 

 “And also, crash = True when controlled vehicles collide with each other.

Ooops, is this not the case already? I'll look into that.”

And it is the case already. 

 

Thanks. 

 

From: ***@***.*** ***@***.***> 
Sent: Sunday, January 23, 2022 12:12
To: 'eleurent/highway-env' ***@***.***>; 'eleurent/highway-env' ***@***.***>
Cc: 'Mention' ***@***.***>
Subject: RE: [eleurent/highway-env] Multi-agent control scenarios (Issue #261)

 

Hi Edouard, 

 

Thanks for your kind reply. This can certainly solve the problem. 

 “And also, crash = True when controlled vehicles collide with each other.

Ooops, is this not the case already? I'll look into that.”

Sorry, I have no idea. Because the 2 vehicles I created are always far from each other. But it’s good if it is already the case. 

Best!

**Comment by eleurent** [comment]
Great :) Closing this, but feel free to open another one if you have more questions
